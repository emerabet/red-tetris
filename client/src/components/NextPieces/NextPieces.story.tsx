import React from 'react';
import { storiesOf } from '@storybook/react';
import NextPieces from './NextPieces';
import * as utils from '../../Utils/constants';

storiesOf('NextPieces', module)
  .add('next pieces', () => (
    <NextPieces
      pieces={[
        utils.SAHPES_T[0],
        utils.SHAPES_I[0],
        utils.SHAPES_Z[0]]}
    />
  ));
