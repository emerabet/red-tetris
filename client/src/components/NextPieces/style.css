.nextPieces {
    width: 100%;
    -webkit-box-shadow: inset 0 0 1px 2px #cfd8dc;
  box-shadow: inset 0 0 1px 2px #cfd8dc;
  margin-top:10px;
  margin-bottom: 20px;
}

.pieceRow {
    display: flex;
    flex-direction: row;
    transform: scaleX(0.7) scaleY(0.7);
}

.pieceColumn {
  padding-top: 10px;
  padding-bottom: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
