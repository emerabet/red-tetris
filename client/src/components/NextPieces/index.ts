import NextPieces from './NextPieces';

export default NextPieces;
