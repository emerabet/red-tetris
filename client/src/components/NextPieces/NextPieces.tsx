import React from 'react';
import RedTetris from '../RedTetris';
import LabeledBox from '../LabeledBox';
import Piece from '../Piece';
import { SHAPES_Z, SHAPES_S, SHAPES_J, SHAPES_L, SAHPES_T, SHAPES_I, SHAPES_O }
from '../../Utils/constants';

import './style.css';

const piecesType:{[name: string]:number[][]} = {
  Z: SHAPES_Z[0],
  S: SHAPES_S[0],
  J: SHAPES_J[0],
  L: SHAPES_L[0],
  T: SAHPES_T[0],
  I: SHAPES_I[0],
  O: SHAPES_O[0],
};

interface NextPiecesProps {
  pieces: string;
  vertical?: boolean;
}

const NextPieces: React.SFC<NextPiecesProps> = (props) => {
  const pcs = props.pieces.split('');
  return (
    <div className="nextPieces">
      <div className={props.vertical ? 'pieceColumn' : 'pieceRow'}>
        {
          pcs.map((piece:string, i:number) => {
            return (
              <Piece key={`piece_${i}`} piece={piecesType[piece]} />
            );
          })
        }
      </div>
    </div>

  );
};

export default NextPieces;
