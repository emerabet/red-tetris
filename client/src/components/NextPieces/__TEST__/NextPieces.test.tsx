import React from 'react';

import renderer from 'react-test-renderer';
import NextPieces from '../NextPieces';

it('renders correctly with defaults', () => {
  const nextPieces = renderer.create(<NextPieces
    pieces="ZOL"
     />).toJSON();
  expect(nextPieces).toMatchSnapshot();
});

it('renders correctly with defaults vertical', () => {
  const nextPieces = renderer.create(<NextPieces
    pieces="ZOL"
    vertical
     />).toJSON();
  expect(nextPieces).toMatchSnapshot();
});