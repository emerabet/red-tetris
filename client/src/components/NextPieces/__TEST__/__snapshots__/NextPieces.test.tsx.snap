// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="pieceRow"
>
  <div
    className="nextPiece"
  >
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell1"
      />
      <div
        className="boardCell cell1"
      />
      <div
        className="boardCell cell0"
      />
    </div>
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell1"
      />
      <div
        className="boardCell cell1"
      />
    </div>
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell0"
      />
    </div>
  </div>
  <div
    className="nextPiece"
  >
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell7"
      />
      <div
        className="boardCell cell7"
      />
    </div>
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell7"
      />
      <div
        className="boardCell cell7"
      />
    </div>
  </div>
  <div
    className="nextPiece"
  >
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell4"
      />
    </div>
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell4"
      />
      <div
        className="boardCell cell4"
      />
      <div
        className="boardCell cell4"
      />
    </div>
    <div
      className="boardRow"
    >
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell0"
      />
      <div
        className="boardCell cell0"
      />
    </div>
  </div>
</div>
`;
